components/hab/src/command/pkg/env.rs

```diff
@@ -12,18 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::collections::HashMap;
 use std::path::Path;
 
 use hcore::package::{PackageIdent, PackageInstall};
 
 use error::Result;
 
-// TODO: This needs a windows compatible version
 pub fn start(ident: &PackageIdent, fs_root_path: &Path) -> Result<()> {
     let pkg_install = PackageInstall::load(ident, Some(fs_root_path))?;
     let env = pkg_install.environment_for_command()?;
+    render_environemt(env)?;
+    Ok(())
+}
+
+#[cfg(not(windows))]
+fn render_environemt(env: HashMap<String, String>) -> Result<()> {
     for (key, value) in env.into_iter() {
         println!("export {}=\"{}\"", key, value);
     }
     Ok(())
 }
+
+#[cfg(windows)]
+fn render_environemt(env: HashMap<String, String>) -> Result<()> {
+    use hcore::fs::FS_ROOT_PATH;
+    use std::env;
+    use std::path::PathBuf;
+
+    let mut paths: Vec<PathBuf> = match env.get("PATH") {
+        Some(path) => env::split_paths(&path).collect(),
+        None => vec![],
+    };
+    for i in 0..paths.len() {
+        if paths[i].starts_with("/") {
+            paths[i] = Path::new(&*FS_ROOT_PATH).join(paths[i].strip_prefix("/").unwrap());
+        }
+    }
+    let joined = env::join_paths(paths)?;
+    let mut mut_env = env.clone();
```

Comment:
Instead of cloning the HashMap, it might be better to just do a `println!` of the modified `PATH`, and then print out the rest of the unmodified map, filtering out the unmodified `PATH` entry.

That way we don't do any unnecessary allocations, and we also more clearly drive home what the key difference is between the two implementations.

Reply:
[*jinx*](url
https://github.com/habitat-sh/habitat/pull/5813#discussion_r230834269)